File: algorithms/selection_sort/selectionSort.js
// JavaScript implementation of selection sort

const selectionSort = (inputArr) => { 
    // Find the length of the array
    let n = inputArr.length;
    
    // Traverse the subarray of each element in the array
    for(let i = 0; i < n; i++) {
        // Finding the smallest number in the subarray
        let min = i;
        for(let j = 0; j < n; j++){
            if(inputArr[j] < inputArr[min]) {
                min=j; 
            }
         }
         // If the current minimum index is not as previous minumum index
         // Swap the values
         if (min != i) {
             // Swapping the elements
             let tmp = inputArr[i]; 
             inputArr[i] = inputArr[min];
             inputArr[min] = tmp;      
        }
    }
    // Return the sorted Array 
    return inputArr;
}

const printArray = (sortedArray) => {
    sortedArray.forEach((el) => {
        console.log(el + " ");
    })
}
// Input Array
let inputArr = [25, 12, 10, 48, 19, 75, 54];

// Calling selectionSort on input Array
let sortedArray = selectionSort(inputArr);

// Calling printArray of print the sorted Array
printArray(sortedArray);

File: algorithms/selection_sort/selection_sort.py
# Python selection sort algorithm
def selection_sort(arr):

    arr_length = len(arr)

    for i in range(arr_length):

        # index for minimum array valuee
        min_index = i

        for j in range(i + 1, arr_length):
            # traverse and find minimum value through remaining array
            if arr[min_index] > arr[j]:
                min_index = j

        # swap elements for current element and found minimum element
        arr[i], arr[min_index] = arr[min_index], arr[i]


# test code
arr = [int(element) for element in input('Enter array:').split()]
selection_sort(arr)
print('Sorted list: ' + str(arr))
